R/query_search.R

```diff
@@ -231,8 +231,14 @@ query_eval_test_binary <- function(op, a, b) {
   if (length(a) == 0 || length(b) == 0) {
     return(logical(0))
   }
-  vlapply(Map(function(a, b) !is.null(a) && !is.null(b) && op(a, b),
-              a, b, USE.NAMES = FALSE),
+  is_equal <- function(a, b) {
```

Comment:
consider moving this into utils, makes it easier to test and does not need to be an inner function